src/transformers/models/beit/configuration_beit.py

```diff
@@ -183,6 +200,26 @@ def __init__(
         self.auxiliary_concat_input = auxiliary_concat_input
         self.semantic_loss_ignore_index = semantic_loss_ignore_index
 
+        # handle backwards compatibility
+        if "segmentation_indices" in kwargs:
+            logger.warning(
+                "The `segmentation_indices` argument is deprecated and will be removed in a future version, use `out_indices` instead.",
+                FutureWarning,
+            )
+            out_indices = kwargs.pop("segmentation_indices")
```

Comment:
Why would segmentation_indices be passed in as a kwarg? It doesn't seem to be an existing argument for the config file

Reply:
The legacy models on the hub only use the `segmentation_indices` argument: https://huggingface.co/microsoft/beit-base-finetuned-ade-640-640/blob/main/config.json#L327-L332. Hence to make this PR backwards compatible, we only need to set the `out_indices` argument equal to the values of `segmentation_indices`

Reply:
We don't just care about backwards compatibility of the original checkpoints. Anyone who's trained a custom beit model will have `out_indices` here e.g. [this model](https://huggingface.co/hkivancoral/hushem_40x_beit_base_f3/blob/main/config.json). 

As mentioned in [my previous comment](https://github.com/huggingface/transformers/pull/25952/#discussion_r1370656949) - this also has unpredicatble behaviour because of how `out_indices` and `out_features` are updated in the config. This still needs to be safely handled.

Reply:
As mentioned on slack, this breaks backwards compatibility for the SemanticSegmentation models: 

```
config = BeitConfig()
model = BeitForSemanticSegmentation(config)
```

produces a different model with this change.